mcx/sample.py

```diff
@@ -385,22 +386,27 @@ def sample_scan(
     """
     num_samples = rng_keys.shape[0]
 
-    print(f"Draw {num_samples:,} samples from {num_chains:,} chains.", end=" ")
-    start = datetime.now()
-
     @jax.jit
-    def update_scan(carry, key):
+    @progress_bar_scan(rng_keys.shape[0])
+    def update_scan(carry, x):
+        key = x[1]
         state, parameters = carry
         keys = jax.random.split(key, num_chains)
         state, info = jax.vmap(kernel, in_axes=(0, 0, 0))(keys, parameters, state)
         return (state, parameters), (state, info)
 
-    last_state, chain = jax.lax.scan(update_scan, (init_state, parameters), rng_keys)
+    define_tqdm(num_samples)
+    set_description_tqdm(
+        f"Collecting {num_samples:,} samples across {num_chains:,} chains"
+    )
+    last_state, chain = jax.lax.scan(
+        update_scan, (init_state, parameters), (np.arange(rng_keys.shape[0]), rng_keys)
```

Comment:
nitpicking: I think that `num_samples` would be clearer than `rng_keys.shape[0]`

Reply:
yes I agree! I will change this